File: .legacy/c++/0045-jump-game-ii.cpp
class Solution {
public:
    // O(n) time | O(1) space
    int jump(vector<int>& nums) {
        int minJumps = 0;
        int l = 0, r = 0;
        int n = nums.size();
        while (r < n - 1) {
            int jump = 0;
            for (int i = l; i < r + 1; ++i) {
                jump = max(jump, i + nums[i]);
            }
            l = r + 1;
            r = jump;
            ++minJumps;
        }
        return minJumps;
    }
};

File: .legacy/python/0045-jump-game-ii.py
class Solution:
    # O(n) time | O(1) space
    def jump(self, nums: List[int]) -> int:
        res = 0
        l = r = 0
        while r < len(nums) - 1:
            jump = 0
            for i in range(l, r + 1):
                jump = max(jump, i + nums[i])
            l = r + 1
            r = jump
            res += 1
        return res
